Spokestack/SpokestackSpeechRecognizer.swift

```diff
@@ -0,0 +1,199 @@
+            let bodySigned = HMAC<SHA256>.authenticationCode(for: body.data(using: .utf8)!, using: self.apiKey!)
+            let bodySignature = Data(bodySigned).base64EncodedString()
+            self.initalizeStreamMessage = """
+            {"keyId": "
+            """ + self.configuration.apiId + """
+            ", "signature": "
+            """ + bodySignature + """
+            ", "body":
+            """ + " " + bodyDoubleEncoded + """
+            }
+            """
+        } else {
+            self.initalizeStreamMessage = ""
+            self.context.error = SpeechPipelineError.apiKey("Unable to encode apiSecret.")
+            self.context.dispatch(.error)
+        }
+        super.init()
+    }
+
+    private func activate() {
+        self.startStream()
+        self.isActive = true
+    }
+
+    private func deactivate() {
+        // send an empty frame to trigger the asr into sending a final response
+        self.stream(self.emptyFrame)
+        //self.task?.suspend()
+        self.context.isActive = false
+        self.isActive = false
+        self.activation = 0
+        self.context.dispatch(.deactivate)
+        //self.task?.cancel(with: .normalClosure, reason: nil)
+    }
+    
+    private func startStream() {
+        // send auth message
+        self.task?.resume()
+        self.task?.send(URLSessionWebSocketTask.Message.string(self.initalizeStreamMessage)) { error in
+            if let error = error {
+                self.context.error = error
+                self.context.dispatch(.error)
+            }
+        }
+        self.task?.receive() { result in
+            self.handle(result, handleResult: { r in
+                if r.status != "ok" {
+                    self.context.error = SpeechPipelineError.illegalState("Spokestack ASR could not start because its status was \(r.status).")
+                    self.context.dispatch(.error)
+                }
+            })
+        }
+    }
+
+    private func stream(_ frame: Data) {
+        /// - TODO: implement a frame chunking policy to maximize MTU utilization. Should be able to chunk ~50ms of frame data at 16000khz. That's ~1500 bytes, but with TLS and websocket overhead the largest buffer without fragmentation is ~1400 bytes. It would probably make more sense to set the buffer size as a multiple of the audio frame size, so something like 1280 bytes.
+        self.activation += self.configuration.frameWidth
+
+        self.task?.send(URLSessionWebSocketTask.Message.data(frame)) { error in
+            if let error = error {
+                self.context.error = error
+                self.context.dispatch(.error)
+            }
+        }
+        self.task?.receive(completionHandler: self.receive)
+    }
+    
+    private func receive(result: (Result<URLSessionWebSocketTask.Message, Error>)) {
+        self.handle(result, handleResult: { r in
+            if let hypothesis = r.hypotheses.last {
```

Comment:
I believe this should be `r.hypotheses.first`, regardless of what you set the limit to.